misc/formatter/modules/formatter-core/src/main/java/org/ballerinalang/formatter/core/FormattingTreeModifier.java

```diff
@@ -281,7 +282,14 @@ public FunctionDefinitionNode transform(FunctionDefinitionNode functionDefinitio
         MetadataNode metadata = formatNode(functionDefinitionNode.metadata().orElse(null), 0, 1);
         NodeList<Token> qualifierList = formatNodeList(functionDefinitionNode.qualifierList(), 1, 0, 1, 0);
         Token functionKeyword = formatToken(functionDefinitionNode.functionKeyword(), 1, 0);
-        IdentifierToken functionName = formatToken(functionDefinitionNode.functionName(), 0, 0);
+        IdentifierToken functionName;
+        if (functionDefinitionNode.relativeResourcePath().isEmpty() ||
+                functionDefinitionNode.relativeResourcePath().get(0).isMissing()) {
```

Comment:
Why are we not keeping a space for func-name, if the first component of relative path is missing?

Reply:
Fixed.